.github/workflows/static-scan.yaml

```diff
@@ -0,0 +1,33 @@
+name: go-static-analysis
+on: [push, pull_request]
+jobs:
+  golangci:
+    name: Lint
+    runs-on: ubuntu-22.04
+    steps:
+      - name: set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.20.x
+      - name: checkout PR
+        uses: actions/checkout@v2
+      - name: run make lint
+        run: make lint
+  shellcheck:
```

Comment:
please, switch to Makefile targets for shellcheck and hadolint

Reply:
Did so with hadolint, dropped the shellcheck as we don't have any shell scripts in this repo